src/Couchbase.Lite.Shared/API/Sync/ReplicatorConfiguration.cs

```diff
@@ -281,26 +285,29 @@ public bool EnableAutoPurge
         /// <summary>
         /// Gets or sets the Max number of retry attempts. The retry attempts will reset
         /// after the replicator is connected to a remote peer. 
-        /// The default is <c>0</c> (<c>10</c> for a single shot replicator or 
-        /// <see cref="int.MaxValue" /> for a continuous replicator is applied.)
-        /// * <c>10</c> for a single shot replicator or <see cref="int.MaxValue" /> for a 
-        /// continuous replicator is applied when user set MaxAttempts to 0.
-        /// * 0 will be returned when default <c>10</c> for a single shot replicator or 
-        /// <see cref="int.MaxValue" /> for a continuous replicator is applied.
         /// * Setting the value to 1 means that the replicator will try connect once and 
         /// the replicator will stop if there is a transient error.
         /// * Default value is <see cref="Constants.DefaultReplicatorMaxAttemptsSingleShot" />
-        /// for a single shot replicator.
+        /// (<c>10</c>) for a single shot replicator.
         /// * Default value is <see cref="Constants.DefaultReplicatorMaxAttemptsContinuous" />
-        /// for a continuous replicator.
+        /// (<see cref="int.MaxValue" />) for a continuous replicator.
         /// </summary>
         /// <exception cref="ArgumentException">
         /// Throw if set the MaxAttempts to a negative value.
         /// </exception>
         public int MaxAttempts
         {
             get => Options.MaxAttempts;
-            set => _freezer.PerformAction(() => Options.MaxAttempts = value);
+            set 
+            {
+                if (value == 0) { // backward compatible when user set the value to 0
+                    _freezer.PerformAction(() => Options.MaxAttempts = Continuous ? Constants.DefaultReplicatorMaxAttemptsContinuous : Constants.DefaultReplicatorMaxAttemptsSingleShot);
+                } else {
+                    _freezer.PerformAction(() => Options.MaxAttempts = value);
+                }
+
+                _isDefaultMaxAttempt = false;
```

Comment:
Not sure about this logic.  If the user resets to the default max attempts value, shouldn't the default behavior come back?